pkl-codegen-kotlin/src/main/kotlin/org/pkl/codegen/kotlin/KotlinCodeGenerator.kt

```diff
@@ -274,9 +274,27 @@ class KotlinCodeGenerator(
       return methodBuilder.addCode(codeBuilder.build()).build()
     }
 
+    fun nearestNonAbstractAncestorDeclaresCopyMethodWithSameArity(): Boolean {
+      if (properties.isNotEmpty()) {
+        // assertion: pClass declares properties that are generated
+        return false
+      }
+      for (currClass in generateSequence(pClass.superclass) { it.superclass }) {
+        if (!currClass.isAbstract) return true
+        if (currClass.properties.values.any { !it.isHidden }) {
+          // assertion: currClass declares properties that are generated
+          return false
```

Comment:
Does this ever get checked? I put a breakpoint here and ran the entire test suite, and never hit this line. Plus, with your new change, we now don't generate copy methods for abstract classes, right?

Reply:
Good catch. This implementation is neither obvious nor correct, so I rewrote it.